File: app/src/main/java/com/ericneidhardt/dynamicsoundboard/BaseFragment.java
package com.ericneidhardt.dynamicsoundboard;


import android.app.Fragment;
import com.ericneidhardt.dynamicsoundboard.soundmanagement.ServiceManagerFragment;
import com.ericneidhardt.dynamicsoundboard.soundsheet.SoundSheetManagerFragment;

public abstract class BaseFragment extends Fragment
{
	public BaseActivity getBaseActivity()
	{
		return (BaseActivity)this.getActivity();
	}

	public NavigationDrawerFragment getNavigationDrawer()
	{
		return (NavigationDrawerFragment)this.getFragmentManager().findFragmentByTag(NavigationDrawerFragment.TAG);
	}

	public SoundSheetManagerFragment getSoundSheetManagerFragment()
	{
		return (SoundSheetManagerFragment)this.getFragmentManager().findFragmentByTag(SoundSheetManagerFragment.TAG);
	}

	public ServiceManagerFragment getServiceManagerFragment()
	{
		return (ServiceManagerFragment)this.getFragmentManager().findFragmentByTag(ServiceManagerFragment.TAG);
	}
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/base/BaseFragment.kt
package org.neidhardt.dynamicsoundboard.base

import org.neidhardt.android_utils.EnhancedSupportFragment
import org.neidhardt.dynamicsoundboard.soundactivity.SoundActivity


abstract class BaseFragment : EnhancedSupportFragment() {
	val baseActivity: SoundActivity get() = this.activity as SoundActivity
}

